src/server/system_services/pool_controllers.js

```diff
@@ -68,88 +67,48 @@ class ManagedStatefulSetPoolController extends PoolController {
         dbg.log2(`ManagedStatefulSetPoolController::create: Pool ${this.pool_name} k8s configuration is`,
             JSON.stringify(pool_k8s_conf, null, 2));
 
-        await kube_utils.apply_conf(pool_k8s_conf);
-        return yaml_utils.stringify(pool_k8s_conf);
+        await KubeStore.instance.create_secret(secret_k8s_conf);
+        await KubeStore.instance.create_backingstore(pool_k8s_conf);
+        const secret_yaml = yaml.stringify(secret_k8s_conf, 6, 2);
+        const backstore_yaml = yaml.stringify(pool_k8s_conf, 6, 2);
+
+        return `${secret_yaml}---\n${backstore_yaml}`;
     }
 
     async scale(host_count) {
-        const pod_count = (await this._list_pods()).length;
-        const delta = host_count - pod_count;
+        const current_host_count = await this.get_current_volume_number();
+        const delta = host_count - current_host_count;
         if (delta === 0) return;
 
         // scale the stateful set
-        const { _statefulset_name } = this;
         dbg.log0(`ManagedStatefulSetPoolController::scale: scaling ${
             delta > 0 ? 'up' : 'down'
-        } stateful set  ${
-            _statefulset_name
+        } backing store set  ${
+            this.pool_name
         } from ${
-            pod_count
+            current_host_count
         } replicas to ${
             host_count
         } replicas`);
 
         if (host_count > 0) {
-            await kube_utils.patch_resource('StatefulSet', _statefulset_name, { spec: { replicas: host_count } });
+            await KubeStore.instance.patch_backingstore(this.pool_name, { spec: { pvPool: { numVolumes: host_count } } });
```

Comment:
You should return bool and handle the case where the backing store was not patched. for example, the backing store was deleted from k8s. In this case I would try to recreate the backing store (already patched)

Reply:
it will throw exception